app/main.py

```diff
@@ -2,4 +2,57 @@ class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords or [0, 0]
+
+    def go_forward(self, steps: int = 1) -> None:
+        self.coords[1] += steps
+
+    def go_back(self, steps: int = 1) -> None:
+        self.coords[1] -= steps
+
+    def go_right(self, steps: int = 1) -> None:
+        self.coords[0] += steps
+
+    def go_left(self, steps: int = 1) -> None:
+        self.coords[0] -= steps
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight, coords)
+        self.coords = coords or [0, 0, 0]
+
+    def go_up(self, steps: int = 1) -> None:
+        self.coords[2] += steps
+
+    def go_down(self, steps: int = 1) -> None:
+        self.coords[2] -= steps
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            max_load_weight: int,
+            current_load: None,
+            coords: list = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: Cargo) -> None:
+        if self.current_load is None and self.max_load_weight >= cargo.weight:
```

Comment:
We can use `not` here

```suggestion
        if not self.current_load and self.max_load_weight >= cargo.weight:
```